File: concrete_trap/stage3.py

fourth_char = ""
fifth_char = ""
TLS = 0

first = int(input())
second = int(input())
third = int(input())
fourth = int(input())
fifth = int(input())

local10 = 0

for i in range(5):
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)
    fdiv = (third + fourth * first) % 32
    first = second // (2 ** fdiv)
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)
    sdiv = (fifth - fourth * first) % 32
    second = fourth * (2**sdiv)
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)
    tdiv = (first + third * second) % 32
    third = fifth // (2 ** tdiv)
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)
    fodiv = (third - fourth * fifth) % 32
    fourth = second * (2 ** fodiv)
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)
    fidiv = (fourth + second * fifth) % 32
    fifth = third // (2 ** fidiv)
    print(first%65536, second%65536, third%65536, fourth%65536, fifth%65536)

    print()

print(first%65536 + second%65536 + third%65536 + fourth%65536 + fifth%65536)

print(0x7a69)

File: concrete_trap/stage3.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>

uint64_t stage3(int32_t first, int32_t second, int32_t third, int32_t fourth, int32_t fifth) {
    char cVar1;
    char cVar2;
    uint64_t uVar3;
    int64_t in_FS_OFFSET;
    int i = 0; 
    i = 0;
    while (i < 5) {
        cVar1 = (char)fourth;
        first = second >> ((char)third + cVar1 * (char)first & 0x1fU);
        cVar2 = (char)(int32_t)fifth;
        second = fourth << (cVar2 - cVar1 * (char)first & 0x1fU);
        third = (int32_t)fifth >> ((char)first + (char)third * (char)second & 0x1fU);
        fourth = second << ((char)third - cVar1 * cVar2 & 0x1fU);
        fifth = third >> ((char)fourth + (char)second * cVar2 & 0x1fU);
        i = i + 1;
    }
    uVar3 = (uint64_t)((int32_t)fifth + first + second + third + fourth == 0x7a69);
    
    if (uVar3 == 0) {
      puts("Bad failure L");
      exit(0);
    }
    puts("Great");

    
    return uVar3;

}

int main() {}

